addon/helpers/breadcrumb.js

```diff
@@ -0,0 +1,35 @@
+import Helper from '@ember/component/helper';
+import { inject as service } from '@ember/service';
+
+export default class BreadcrumbHelper extends Helper {
+  @service('breadcrumbs') breadcrumbsService;
+
+  breadcrumbId = null;
+
+  get breadcrumbWasAddedBefore() {
+    return Boolean(this.breadcrumbId);
+  }
+
+  compute(breadcrumbTitleParts, optionalData) {
+    let breadcrumbTitle = breadcrumbTitleParts.join(' ');
+
+    let breadcrumbData = {
+      ...optionalData,
+      title: breadcrumbTitle,
+    };
+
+    if (this.breadcrumbWasAddedBefore) {
+      this.breadcrumbsService.updateBreadcrumb(
+        this.breadcrumbId,
+        breadcrumbData
+      );
+    } else {
+      this.breadcrumbId = this.breadcrumbsService.addBreadcrumb(breadcrumbData);
+    }
+  }
+
+  willDestroy() {
+    super.willDestroy();
```

Comment:
```suggestion
    super.willDestroy(...arguments);
```

For completeness sake.